protocol/swap/types.go

```diff
@@ -63,8 +62,18 @@ type Info struct {
 	// (and after Timeout1), the ETH-taker is able to claim, but
 	// after this timeout, the ETH-taker can no longer claim, only
 	// the ETH-maker can refund.
-	Timeout2 *time.Time        `json:"timeout2,omitempty"`
-	statusCh chan types.Status `json:"-"`
+	Timeout2 *time.Time `json:"timeout2,omitempty"`
+
+	// rwMu handles synchronization when LastStatusUpdateTime, Timeout1,
+	// Timeout2 and EndTime are updated. This Info struct is modified by the
+	// maker or taker's swapState go process as the state of the swap
+	// progresses. The swapState go process does not need synchronization when
+	// reading its own changes, but it needs to grab a write lock when modifying
+	// the the structure. Readers from other go-processes only get copies of
+	// this structure. They exclusively use the DeepCopy method to get their
+	// copy, which grabs the read lock ensuring that they always capture the
+	// up-to-date state of this Info struct.
+	rwMu sync.RWMutex
```

Comment:
why did you decide to go with a lock in the info struct instead of having the swap manager `Get` functions be read-only (ie. only return copies) and having swap manager implement `Set` functions which modify the info fields for a given swap?

Reply:
The swap manager's existing locks are for the maps inside the manager. They act globally across all `Info` objects in the ongoing or past swap maps. To manage the state the state of an individual swap, we needed a lock that is specific to that individual `Info` object. The swap manager could maintain a map of locks for the individual `Info` structures, but I thought this solution was cleaner.